public/modules/custom/helfi_rekry_content/helfi_rekry_content.module

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types = 1);
+
+use Drupal\migrate\MigrateSkipRowException;
+
+/**
+ * Format filename from source uri
+ * 
+ * @param string|null
+ *   The source uri
+ * 
+ * @return string|null
+ *   Filename or null
+ */
+function _helfi_rekry_content_filename(string|null $sourceUri = NULL): ?string  {
+  if(!$sourceUri) {
+    return NULL;
+  }
+
+  return strtok(basename($sourceUri), '?');
+}
+
+/**
+ * Get media entity by file id
+ * 
+ * @param string|null
+ *   The file id
+ * 
+ * @return string|null
+ *   Media entity id or null
+ */
+function _helfi_rekry_content_get_media_image(string|null $fid = NULL): ?string {
+  if(!$fid) {
+    return NULL;
+  }
+
+  $entityQuery = \Drupal::entityQuery('media')
+  ->condition('bundle', 'job_listing_image')
+  ->condition('field_media_image.target_id', $fid);
+
+  return reset($entityQuery->execute());
```

Comment:
This will throw deprecation warning on PHP 8.1: 
![image](https://user-images.githubusercontent.com/771113/168233585-aeaa7c17-226f-4370-bd89-2a28453558a0.png)

https://www.php.net/manual/en/function.reset.php